prime-router/docs/design/proposals/0023-condition-to-code-mapping/0023-condition-to-code-mapping.md

```diff
@@ -0,0 +1,229 @@
+
+There are multiple ways to write the logic in the condition filter. The first way will remove any observations that do not pass the condition filter.<br>
+
+### Condition filter logic #1
+
+Example input: [prime-reportstream/prime-router/docs/design/proposals/0023-condition-to-code-mapping/exampleinput.fhir]()
+
+Example condition logic:
+```yaml
+    - name: TEST-RECEIVER
+      externalName: TEST
+      organizationName: TEST
+      topic: full-elr
+      customerStatus: active
+      jurisdictionalFilter:
+        - "(%performerState.exists() and %performerState = 'TEST') or (%patientState.exists() and %patientState = 'TEST')"
+      qualityFilter: []
+      conditionFilter:
+          - "%resource.where(meta.tag.code in (398102009' |'13906002' |'409498004' |'40610006' |'21061004' |'75702008' |'359761005' |'414015000' |'840539006' |'416925005' |'83436008')).exists()"
+      timing:
+        operation: MERGE
+        numberPerDay: 1440 # Every minute
+        initialTime: 00:00
+```
+Example output: [prime-reportstream/prime-router/docs/design/proposals/0023-condition-to-code-mapping/exampleoutput1.hl7]()
+
+This will result in a message with potentially missing results in the event that there is a missing mapping, and also presents difficulties in dealing with culture results as it would require all the LOINCS for all possible micro cultures to be mapped.
+
+### Condition filter logic #2
+
+The second way to write the condition logic will pass the entire bundle if a match is found for one observation.
+
+Example input: [prime-reportstream/prime-router/docs/design/proposals/0023-condition-to-code-mapping/exampleinput.fhir]()
+
+```yaml
+    - name: TEST-RECEIVER
+      externalName: TEST
+      organizationName: TEST
+      topic: full-elr
+      customerStatus: active
+      jurisdictionalFilter:
+        - "(%performerState.exists() and %performerState = 'TEST') or (%patientState.exists() and %patientState = 'TEST')"
+      qualityFilter: []
+      conditionFilter:
+          - "bundle.entry.ofType(Observation).where(meta.tag.code in (398102009' |'13906002' |'409498004' |'40610006' |'21061004' |'75702008' |'359761005' |'414015000' |'840539006' |'416925005' |'83436008')).exists()"
+      timing:
+        operation: MERGE
+        numberPerDay: 1440 # Every minute
+        initialTime: 00:00
+```
+Example output: [prime-reportstream/prime-router/docs/design/proposals/0023-condition-to-code-mapping/exampleoutput2.hl7]()
+
+This will potentially pass along results for conditions that are not desired by the receiver in the event the message/bundle represents a multiplex test where the receiver only desires some of the test results (e.g. if NY wanted to receive FLU A and FLU B results but not covid results and ReportStream received a messages where all three were represented, all three would be sent). This is likely an edge case that will not be commonly encountered and the risk of missing results by going with option #1 is much higher.
```

Comment:
Just want to call out that there is already some handling around this and might make sense to add that to the example here?  That logic works by adding an `Endpoint` resource to the bundle that indicates which observations the receiver would like to get.

Specifically, [this](https://github.com/CDCgov/prime-reportstream/blob/70e9cc5bd4260116cbaad253ddf8080cacf7701c/prime-router/src/main/kotlin/fhirengine/engine/FHIRTranslator.kt#L184) code will remove any observations for a particular receiver based on changes to the bundle that occur [here](https://github.com/CDCgov/prime-reportstream/blob/70e9cc5bd4260116cbaad253ddf8080cacf7701c/prime-router/src/main/kotlin/fhirengine/utils/FHIRBundleHelpers.kt#L65)

Reply:
Getting back to this. Took me awhile to parse how that was working. I'll add some language calling that out